packages/flutter/test/widgets/editable_text_utils.dart

```diff
@@ -139,3 +139,21 @@ class OverflowWidgetTextEditingController extends TextEditingController {
     );
   }
 }
+
+// Trigger MediaQuery to update itself based on the View, which is not
+// recreated between tests. This is necessary when changing something on
+// TestPlatformDispatcher and expecting it to be picked up by MediaQuery.
+void updateMediaQuery(WidgetTester tester) {
+  expect(find.byType(MediaQuery), findsOneWidget);
+  final WidgetsBindingObserver widgetsBindingObserver =
+      tester.state(
+            find.ancestor(
+              of: find.byType(MediaQuery),
+              matching: find.byWidgetPredicate(
+                (Widget w) => '${w.runtimeType}' == '_MediaQueryFromView',
+              ),
+            ),
+          )
+          as WidgetsBindingObserver;
+  widgetsBindingObserver.didChangeMetrics();
```

Comment:
Does anyone know of a better way to do this? MediaQuery.fromView gets its values set from the view once per run of `flutter test`. When I change isSupported on the view, MediaQuery does not reflect the updated value unless I call something like didChangeMetrics on it.

Reply:
@Renzo-Olivares had an idea to insert another MediaQuery into the tree and set its value of supportsShowingSystemContextMenu. I tried it and it didn't seem to work. SystemContextMenu is built in contextMenuBuilder, and the BuildContext it gets is from its OverlayEntry, which is near the root of the tree. So unfortunately when it looks up the nearest MediaQuery, it misses the one that I put in the tree.

Reply:
Another thing to try from @Renzo-Olivares was changing the AppLifecycleState to get the view to rebuild. This succeeded in getting _MediaQueryFromView.didUpdateWidget to be called, but the logic in that method does not update the view data unless the entire view changes.

I suspect that might be a bug, so I opened an issue: https://github.com/flutter/flutter/issues/165519.

Reply:
I would consider making this method private if we intend to change it, but I also think its fine as is since its in the framework tests.

Reply:
I made it private, I think that's probably a good call despite the duplication. Hopefully we can fix the bug and then remove it.